software/base/src/main/java/org/apache/brooklyn/entity/software/base/SoftwareProcess.java

```diff
@@ -147,6 +147,9 @@
     @SetFromFlag("runDir")
     AttributeSensorAndConfigKey<String,String> RUN_DIR = BrooklynConfigKeys.RUN_DIR;
 
+    ConfigKey<Boolean> ADD_OPEN_PORTS_EFFECTOR = ConfigKeys.newBooleanConfigKey("effector.add.openPorts",
```

Comment:
My idea was that the class `AddEffector` is a generic way to attach effectors to entities in yaml. Then no need for an ad-hoc flag to attach the effector.
Looking at the implementation though it's not straightforward to use it in YAML currently so ignore.